tests/test_signal_generator.cpp

```diff
@@ -0,0 +1,169 @@
+
+#define TEST_AMPLITUDE 100.0
+#define TEST_FREQUENCY 30.0
+#define TEST_COSINE false
+#define TEST_TIMESTAMP 1000000000
+#define TEST_FRAMID     "Teste 1"
+#define TEST_VALUE      20.0
+#define TEST_RETURNED_VALUE_POS 1.0
+#define TEST_RETURNED_VALUE_NEG 0.0
+
+static void
+__assert_and_run_state_machine(hyro::StateMachine &sm,
+                               const hyro::ComponentConfiguration &conf,
+                               const hyro::ConnectionConfiguration &endpoint)
+{
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.init(conf));
+
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.start());
+
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.connect(endpoint));
+
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.check());
+}
+
+using namespace signal_generator;
+using namespace digital_converter;
+
+namespace hyro
+{
+namespace test
+{
+
+TEST(ExampleTest, SignalGeneratorComponentCheck)
+{
+  //Configuration
+  auto generator_configuration = "{"
+                                 "outputs: {"
+                                 //"analog_output: { protocol: 'api', ip: '0.0.0.0',   port: '50051' },"
+                                 "analog_output: { protocol: 'api'},"
+                                 "fix_dynamic: { protocol: 'api'},"
+                                 "}"
+                                 "}";
+
+  //IntGeneratorComponent state machine
+  StateMachine sm(std::make_shared<SignalGeneratorComponent>("/generator"_uri));
+
+  auto result = sm.init(ComponentConfiguration(generator_configuration));
+  ASSERT_EQ(Result::RESULT_OK, result);
+
+  result = sm.start();
+  ASSERT_EQ(Result::RESULT_OK, result);
+
+  result = sm.connect(ConnectionConfiguration());
+  ASSERT_EQ(Result::RESULT_OK, result);
+
+  result = sm.check();
+  ASSERT_EQ(Result::RESULT_OK, result);
+
+  //Fake input channels
+  auto AnalogOut = std::make_shared<FakeInput<SignalMsgs>>("AnalogOut"_uri, "api", "/generator/analog_output");
+  ASSERT_TRUE(AnalogOut->connect());
+
+  DynamicPropertyAccess dynamic_property_access("/generator"_uri);
+
+  auto ret = dynamic_property_access.set<bool>("cosine", TEST_COSINE);
+  ASSERT_EQ(ret, true);
+  ASSERT_TRUE(dynamic_property_access.set<float>("amplitude", TEST_AMPLITUDE));
+  ASSERT_TRUE(dynamic_property_access.set<float>("frequency", TEST_FREQUENCY));
+
+  bool cosine;
+  dynamic_property_access.get<bool>("cosine", cosine);
+  ASSERT_EQ(cosine, TEST_COSINE);
+
+  float amp, freq;
+  dynamic_property_access.get<float>("amplitude", amp);
+  ASSERT_EQ(amp, TEST_AMPLITUDE);
+  dynamic_property_access.get<float>("frequency", freq);
+  ASSERT_EQ(freq, TEST_FREQUENCY);
+
+  sm.update();
+
+  auto value_analog_out = std::shared_ptr<const SignalMsgs>();
+
+  auto return_value = AnalogOut->receive(value_analog_out, 500ms);
+  ASSERT_EQ(ReceiveStatus::RECEIVE_OK, return_value);
+  ASSERT_GE(value_analog_out->value,(-TEST_AMPLITUDE));
+  ASSERT_LE(value_analog_out->value,TEST_AMPLITUDE);
+}
+
+TEST(ExampleTest, DigitalConverterComponentCheck)
```

Comment:
The same for this test case. It is really good programmed. I just add the `ASSERT_NEAR` here to check the output of the output channel.

Reply:
In this case it was done. I checked for values above and below the threshold.

Reply:
Ok @rfcarneiro. Let's keep like that! Anyway, the testing is for introduce you to GTEST library.

Reply:
:+1: